lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
```

Comment:
Returning `undefined` allows us differentiate "lack of _headers set" (i.e., possibly flushed) from "empty headers set."